tests/blockchain/eth/contracts/lib/test_signature_verifier.py

```diff
@@ -52,36 +52,27 @@ def test_recover(testerchain, signature_verifier):
 
     # Generate Umbral key and extract "address" from the public key
     umbral_privkey = UmbralPrivateKey.gen_key()
+    umbral_pubkey = umbral_privkey.get_pubkey()
     umbral_pubkey_bytes = umbral_privkey.get_pubkey().to_bytes(is_compressed=False)
-    signer_address = bytearray(testerchain.interface.w3.solidityKeccak(['bytes32'], [umbral_pubkey_bytes[1:]]))
+    signer_address = keccak_digest(umbral_pubkey_bytes[1:])
     signer_address = to_normalized_address(signer_address[12:])
 
-    # Sign message using SHA-256 hash (because only 32 bytes hash can be used in the `ecrecover` method)
-    cryptography_priv_key = umbral_privkey.to_cryptography_privkey()
-    signature_der_bytes = cryptography_priv_key.sign(message, ec.ECDSA(hashes.SHA256()))
-    signature = Signature.from_bytes(signature_der_bytes, der_encoded=True)
+    # Sign message
+    signer = Signer(umbral_privkey)
+    signature = signer(message)
 
     # Get recovery id (v) before using contract
-    # If we don't have recovery id while signing than we should try to recover public key with different v
+    # If we don't have recovery id while signing then we should try to recover public key with different v
     # Only the correct v will match the correct public key
-    # First try v = 0
-    recoverable_signature = bytes(signature) + bytes([0])
-    pubkey_bytes = coincurve.PublicKey.from_signature_and_message(recoverable_signature, message_hash, hasher=None)\
-        .format(compressed=False)
-    if pubkey_bytes != umbral_pubkey_bytes:
-        # Extracted public key is not ours, that means v = 1
-        recoverable_signature = bytes(signature) + bytes([1])
-        pubkey_bytes = coincurve.PublicKey.from_signature_and_message(recoverable_signature, message_hash, hasher=None)\
-            .format(compressed=False)
-
-    # Check that recovery was ok
-    assert umbral_pubkey_bytes == pubkey_bytes
+    v = get_signature_recovery_value(message, signature, umbral_pubkey)
+    recoverable_signature = bytes(signature) + v
+
     # Check recovery method in the contract
     assert signer_address == to_normalized_address(
         signature_verifier.functions.recover(message_hash, recoverable_signature).call())
 
     # Also numbers 27 and 28 can be used for v
-    recoverable_signature = recoverable_signature[0:-1] + bytes([recoverable_signature[-1] + 27])
+    recoverable_signature = recoverable_signature[:-1] + bytes([recoverable_signature[-1] + 27])
```

Comment:
This is beginning to smell like we need to probably add this functionality to `Signature`.

Reply:
Can you clarify this? This line in particular does something very specific for the test, nothing actually useful.

Reply:
Why not add recovery methods to the `Signature` object? Why add them to `utils`?

Reply:
Mainly because `Signature` is from `umbral`, although I agree that from an OOP perspective it makes sense to do it in `Signature`. Should I do this in Umbral instead?